v2g-liberty/rootfs/root/appdaemon/apps/v2g_liberty/modbus_evse_client.py

```diff
@@ -1042,90 +1123,121 @@ async def __base_polling(self, kwargs):
     #                   MODBUS RELATED FUNCTIONS                         #
     ######################################################################
 
-    async def __update_charger_connection_state(self, is_alive: bool = True):
+    async def __update_charger_connection_state(self, is_alive: bool):
         keep_alive = {"keep_alive": get_local_now()}
         msg = "Successfully connected" if is_alive else "Connection error"
-        await self.__update_state(
+        await self.__update_ha_entity(
             entity_id="sensor.charger_connection_status",
-            state=msg,
+            new_value=msg,
             attributes=keep_alive,
         )
 
     async def __force_get_register(
-        self, register: int, min_value: int, max_value: int
+        self,
+        register: int,
+        min_value_at_forced_get: int,
+        max_value_at_forced_get: int,
+        min_value_after_forced_get: int = None,
+        max_value_after_forced_get: int = None,
     ) -> int:
-        """When a 'realtime' reading from the modbus server is needed that is expected
-        to be between min_value/max_value. We keep trying to get a reading that is within min_value - max_value range.
-
-        Of course there is a timeout. If this is reached we expect the modbus server to have crashed,
-        and we notify the user.
-
+        """
+        When a 'realtime' reading from the modbus server is needed, as opposed to
+        stored value from polling.
+        It is expected to be between min_value_at_forced_get/max_value_at_forced_get.
+        Keep trying to get a reading that is within min_value_at_forced_get - max_value_at_forced_get range or
+        a timeout has been reached.
+        After the timeout the value is checked against other min/max values. This is
+        a wider range.
+        This is aimed at the SoC, this is expected to be between 2 and 97%, but at
+        timeout 1% to 100% is acceptable.
+
+        If the value is not in the wider acceptable range at timeout we assume
```

Comment:
I can only assume that this is based on real-life experience. However, even the comment seems awefully complicated to me, which suggests the code will be also complicated, and therefore by extension have a high risk of complicated bugs.

Reply:
It indeed is based on the (long) experience with the Quasar charger.
So, we need complex code to deal with some strange behaviour of the charger.
I agree that the comments are not helpful so i removed some.